### DISCUSSION TITLE
Running Zephyr RTOS on STM32F303CBT6

### DISCUSSION THREAD

**Discussion by sbendt** [zephyrproject-rtos/zephyr#58781] [Q&A] [answered]
I have a custom board with an STM32F303C**B**T6 MCU with an external 16 MHz oscillator, but I am unfortunately not able to get Zephyr RTOS to successfully boot.

From the KConfig options, it does not look like the specific variant of F303 is supported. It does have the following configs:

CONFIG_SOC_STM32F303X**8**
CONFIG_SOC_STM32F303X**C**
CONFIG_SOC_STM32F303X**E**
In addition, the device trees to include is also:

zephyr\dts\arm\st\f3\stm32f303X**8**.dtsi
zephyr\dts\arm\st\f3\stm32f303X**c**.dtsi
zephyr\dts\arm\st\f3\stm32f303X**e**.dtsi

Both the config and dtsi are missing the stm32f303Xb variants.

I tried to put my own dtsi file together, but so far without any luck. 

When using the attached dtsi, the debugger gets stuck in "LL_RCC_HSE_IsReady" defined in modules/hal/stm32/stm32cube/stm32f3xx/drivers/include/stm32f3xx_ll_rcc.h

Debugger is an STM32 programmer connected by SWD to the board.

Any recommendations to get the specific MCU working with Zephyr RTOS?

**Comment by erwango** [comment]
It is likely that the issue you're facing is not linked to the SoC variant, but to the clock configuration.
Can you provide the clock configuration you're using (extracted from board .dts) ?

**Comment by sbendt** [comment]
Hi,

The intended clock configuration should be something like this:
![ClockConfig](https://github.com/zephyrproject-rtos/zephyr/assets/135331864/a85b5f2e-6ba5-4f12-bea3-6612cb8830b1)

The clock configuration I have in the dts file is as below, but is retting stuck during initialization before main().

&clk_lsi {
	status = "disabled";
};

&clk_hse {
	hse-bypass;
	clock-frequency = <DT_FREQ_M(16)>; 
	status = "okay";
};

&pll {
	prediv = <2>;
	mul = <9>;
	clocks = <&clk_hse>;
	status = "okay";
};

&rcc {
	clocks = <&pll>;
	clock-frequency = <DT_FREQ_M(16)>;
	ahb-prescaler = <1>;
	apb1-prescaler = <2>;
	apb2-prescaler = <1>;
};

&rtc {
	clocks = <&rcc STM32_CLOCK_BUS_APB1 0x10000000>,
		 <&rcc STM32_SRC_LSI RTC_SEL(2)>;
	status = "okay";
};

**Reply by erwango** [reply]
I see at least 2 issues.

1) Core clock freq should be 72 
You're defining HSE freq to 16 Mhz.
Core clock-freq (clock-frequency in &rcc node) = HSE freq / prediv * mul = 72Mhz (not 16).

2) LSI should status should be set as "enabled" since it is defined as domain clock for RTC

Then you need to verify that HSE settings are matching your board.

**Reply by sbendt** [reply]
Thanks! I did a few modifications based on you input. It is no longer stuck in the LL_RCC_HSE_IsReady, which is great!
I did remove the hse-bypass as well in order to get it working. As far as I know, it means it is using the external oscillator?

It does however seem to affect the UART's, since transmitted data is not being received. Any idea what can affect that?

&clk_lsi {
	status = "okay";
};

&clk_hse {
	//hse-bypass;
	clock-frequency = <DT_FREQ_M(16)>; 
	status = "okay";
};

&pll {
	prediv = <2>;
	mul = <9>;
	clocks = <&clk_hse>;
	status = "okay";
};

&rcc {
	clocks = <&pll>;
	clock-frequency = <DT_FREQ_M(72)>;
	ahb-prescaler = <1>;
	apb1-prescaler = <2>;
	apb2-prescaler = <1>;
};

&rtc {
	clocks = <&rcc STM32_CLOCK_BUS_APB1 0x10000000>,
		 <&rcc STM32_SRC_LSI RTC_SEL(2)>;
	status = "okay";
};

**Reply by erwango** [reply, marked as answer]
> It does however seem to affect the UART's, since transmitted data is not being received. Any idea what can affect that?

Was it working previously ?

If you didn't touch UART clock settings, it should work out of the box, as UART clock prescaler is computed automatically based on bus clock.
If it didn't work ever. I'd suggest to check pinctrl settings and make sure correct uart node is enabled.

**Reply by sbendt** [reply]
Sorry about the confussion - turns out my USB FTDI thingy was malfunctioning. Replacing it solved the problem.

Thank you very much for your support - so far everything checks out : )
